jaraco/develop/git.py

```diff
@@ -109,29 +107,75 @@ def path(self):
 
 class Project(str):
     """
-    >>> p = Project.parse('foo-project [tag1] [tag2]')
+    >>> p = Project.parse('foo-project [tag1] [tag2] (zero defect, coherent software)')
     >>> p
     'foo-project'
     >>> p.tags
     ['tag1', 'tag2']
     >>> p.spec
     'foo-project [tag1] [tag2]'
+    >>> p.topics
+    ['zero defect', 'coherent software']
     """
 
     pattern = re.compile(r'(?P<name>\S+)\s*(?P<rest>.*)$')
     tags: list[str] = []
-
-    def __new__(self, value, **kwargs):
-        return super().__new__(self, value)
+    cache: dict[str, Project] = {}
+
+    def __new__(cls, value, **kwargs):
+        # Down-cast to a string early.
+        value = sys.intern(str(value))
+        try:
+            return cls.cache[value]
+        except KeyError:
+            new = super().__new__(cls, value)
+            cls.cache[new] = new
+            return new
 
     def __init__(self, value, **kwargs):
-        vars(self).update(kwargs)
+        vars(self).update({'topics': [], **kwargs})
```

Comment:
Note how I've moved the default for `tags` to a class property. probably we should do the same with topics (if keeping them).

Reply:
I think you opted for not keeping them in https://github.com/jaraco/dotfiles/pull/2.
I'll rework this PR accordingly.